src/console2.sol

```diff
@@ -1543,4 +1590,1373 @@ library console2 {
         _sendLogPayload(abi.encodeWithSignature("log(address,address,address,address)", p0, p1, p2, p3));
     }
 
+    // Log Functions with Color functionality.
+        function log(Console storage self, uint256 p0) internal view {
+        _sendLogPayload(abi.encodeWithSignature("log(string,uint256,string)", self._color, p0, CRESET));
```

Comment:
Should we reset the color after each `log` call? I suspect that's the normal UX that people would expect, i.e. change behavior:

```solidity
// From this:
console.red().log("Test");
console.log("This will still be red");

// To this:
console.red().log("Test");
console.log("This is back to the normal color");
```

Reply:
I originally developed it that way, resetting the color after each call. However it should be noted that due to the way console logging is implemented right now through a static call only 32 bytes of data can be logged for each line.

So the color set and reset will consume part of this 32 bytes. So less data can be logged on a single line. If we do it through 3 static calls so users can log 32 bytes even after setting colors there will be a breakline every time a color is set/reset so this should require some rework on forge/anvil.

However shouldn't be a big deal, so I agree on resetting the color after each call, I can modify it and push a new commit if you want me to do so.

Reply:
> only 32 bytes of data can be logged for each line...So less data can be logged on a single line. 

Hmm I don't think this is correct? `_sendLogPayload` takes an arbitrary `bytes memory` so logs can be any length. Given that resetting the color after each call seems like it shouldn't be an issue